src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -204,7 +204,7 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
   }, [appName, chainId, closeSignMessageModal, closeTxModal, communicator, signMessageModalState, txModalState])
 
   useEffect(() => {
-    const unsubFns = [SafeMsgEvent.PROPOSE, SafeMsgEvent.CONFIRM_PROPOSE].map((event) => {
+    const unsubFns = [SafeMsgEvent.SIGNATURE_PREPARED].map((event) => {
```

Comment:
You no longer need loop over this. It should also preserve type as well meaning you can remove some of the logic within the callback.